OpenSubtitlesHandler/Util.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            var hash = ComputeMovieHash(stream);
+            return Convert.ToHexString(hash).ToLowerInvariant();
+        }
+
+        /// <summary>
+        /// Serialize object into JSON
+        /// </summary>
+        /// <returns>JSON string of the object</returns>
+        public static string Serialize(object o)
+        {
+            return JsonSerializer.Serialize(o);
+        }
+
+        /// <summary>
+        /// Deserialize object from JSON
+        /// </summary>
+        /// <returns>Deserialized object</returns>
+        public static T Deserialize<T>(string str)
+        {
+            return JsonSerializer.Deserialize<T>(str, SerializerOpts);
+        }
+
+        /// <summary>
+        /// Compute hash of specified movie stream
+        /// </summary>
+        /// <returns>Hash of the movie</returns>
+        private static byte[] ComputeMovieHash(Stream input)
+        {
+            using (input)
+            {
+                var streamSize = input.Length;
+                var lHash = streamSize;
+
+                long i = 0;
+                byte[] buffer = new byte[sizeof(long)];
+
+                while (i < 65536 / sizeof(long) && (input.Read(buffer, 0, sizeof(long)) > 0))
+                {
+                    i++;
+                    lHash += BitConverter.ToInt64(buffer, 0);
+                }
+
+                input.Position = Math.Max(0, streamSize - 65536);
+                i = 0;
+
+                while (i < 65536 / sizeof(long) && (input.Read(buffer, 0, sizeof(long)) > 0))
+                {
+                    i++;
+                    lHash += BitConverter.ToInt64(buffer, 0);
+                }
+
+                byte[] result = BitConverter.GetBytes(lHash);
+                Array.Reverse(result);
+
+                return result;
+            }
+        }
+
+        internal static async Task<(string, Dictionary<string, string>, HttpStatusCode)> SendRequestAsync(string url, HttpMethod method, string body, Dictionary<string, string> headers, CancellationToken cancellationToken)
+        {
+            if (!HttpClient.DefaultRequestHeaders.Contains("User-Agent"))
+            {
+                if (string.IsNullOrWhiteSpace(_version))
+                {
+                    throw new Exception("Missing plugin version");
+                }
+
+                var ua = $"Jellyfin-Plugin-OpenSubtitles/{_version}";
+
+                HttpClient.DefaultRequestHeaders.Add("User-Agent", ua);
+            }
+
+            HttpContent content = null;
+            if (method != HttpMethod.Get && !string.IsNullOrWhiteSpace(body))
+            {
+                content = new StringContent(body, Encoding.UTF8, "application/json");
+            }
+
+            var request = new HttpRequestMessage
+            {
+                Method = method,
+                RequestUri = new Uri(url),
+                Content = content
+            };
+
+            // docs say alphabetical order improves speed
```

Comment:
I think you've misread the docs (it happens). https://opensubtitles.stoplight.io/docs/opensubtitles-api/open_api.json#performances

GET parameters should be in alphabetical order, not the headers.

Reply:
Fixed in bd3f223096cb560a9c416cba5337e9d970a9adfe